### ISSUE TITLE
Enable Chronos-2 to perform native multivariate prediction

### ISSUE THREAD

**Issue by RDR2Blackwater** [awslabs/autogluon#5510]
## Description
As mentioned in the title, the `Chronos2Pipeline.predict_df()` method supports native multivariate inputs:
```
def predict_df(
    self,
    df: “pd.DataFrame”,
    future_df: “pd.DataFrame | None” = None,
    id_column: str = “item_id”,
    timestamp_column: str = “timestamp”,
    target: str | list[str] = “target”,
    ...
```
However, in the current AutoGluon implementation, inheriting from the `AbstractTimeSeriesModel` class prevents this, as it only supports `str`-type target inputs:
```
class AbstractTimeSeriesModel(TimeSeriesModelBase, TimeSeriesTunable, metaclass=ModelRegistry):
    “”"Abstract base class for all time series models that take historical data as input and
    make predictions for the forecast horizon.
    “”"

    ag_priority: int = 0

    def __init__(
        self,
        path: str | None = None,
        name: str | None = None,
        hyperparameters: dict[str, Any] | None = None,
        freq: str | None = None,
        prediction_length: int = 1,
        covariate_metadata: CovariateMetadata | None = None,
        target: str = “target”,
        ...
```
I know that directly overriding the entire `__init__` method to support `List[str]` input would introduce many issues, like the `TimeSeriesScoring` class's handling of `target`. However, without doing so, Chronos-2's joint prediction ability cannot be fully demonstrated when dealing with multivariate problems.

The current solution is to initialize a `TimeSeriesPredictor` class for each target, as described in [tabular multilabel practice](https://auto.gluon.ai/stable/tutorials/tabular/advanced/tabular-multilabel.html), just like the other time series models. Or directly instantiate the `Chronos2Model` class and call its methods for prediction and fine-tuning. But we still need better solutions.

**Comment by RDR2Blackwater** [comment]
Btw, the default batch size is set to 256, consistent with Chronos-bolt, and Joint Prediction has been enabled by default. However, according to Chronos-2's implementation of Joint Prediction, when cross learning is enabled, a larger batch size isn't necessarily better:

> ...Very large batch sizes may not provide benefits as they deviate from the maximum group size used during pretraining. For optimal results, consider using a batch size around 100 (as used in the paper).

The Notebook specifies a batch size of 100 instead of 256:
```
# Example: Enable cross-learning for joint prediction
# This assigns the same group ID to all time series, allowing information sharing
joint_pred_df = pipeline.predict_df(
    context_df,
    prediction_length=24,
    quantile_levels=[0.1, 0.5, 0.9],
    cross_learning=True,  # Enable cross-learning
    batch_size=100,
)
```

Or have you conducted a new round of testing that demonstrates `batch_size=256` performs similarly to/better than a `batch _ize=100` when cross learning is enabled?

**Comment by shchur** [comment]
Hi @RDR2Blackwater, thanks for opening the issue. Enabling native multivariate support (`target: list[str]`) will require a major refactor of all components of the `autogluon.timeseries` module, so we currently don't plan to work on this. 

However, given how cross-learning/multivariate forecasting is implemented in Chronos-2, the users can get identical results by simply "melting" the wide DataFrame into a long DataFrame, something like the following
```python
def multivariate_to_univariate(
    df: pd.DataFrame,
    item_id_col: str,
    time_col: str,
    target_cols: list[str],
    target_col: str = "target",
) -> pd.DataFrame:
    out = df.melt(
        id_vars=[item_id_col, time_col],
        value_vars=target_cols,
        var_name="__var",
        value_name=target_col,
    )

    # New item_id will be "{old_item_id}_{target_column}"
    out[item_id_col] = out[item_id_col].astype(str) + "_" + out["__var"]

    return (
        out
        .drop(columns="__var")
        .sort_values([item_id_col, time_col])
        .reset_index(drop=True)
    )
```

This works because with `cross_learning=True`, Chronos-2 does not differentiate between individual univariate items, or multiple multivariate dimensions of a single item. In other words, the following two inputs will produce identical results
item_id | timestamp | X | Y
-- | -- | -- | --
A | 1 | 10 | 20
A | 2 | 11 | 21
A | 3 | 12 | 22
B | 1 | 30 | 40
B | 2 | 31 | 41
B | 3 | 32 | 42

item_id | timestamp | target
-- | -- | --
A_X | 1 | 10
A_X | 2 | 11
A_X | 3 | 12
A_Y | 1 | 20
A_Y | 2 | 21
A_Y | 3 | 22
B_X | 1 | 30
B_X | 2 | 31
B_X | 3 | 32
B_Y | 1 | 40
B_Y | 2 | 41
B_Y | 3 | 42


In our internal benchmarks we found `batch_size=256` to not result in any noticeable accuracy decrease compared to `batch_size=100`, while resulting in faster inference speed. That's why we decided to keep this value as the default

**Comment by RDR2Blackwater** [comment]
> Hi [@RDR2Blackwater](https://github.com/RDR2Blackwater), thanks for opening the issue. Enabling native multivariate support (`target: list[str]`) will require a major refactor of all components of the `autogluon.timeseries` module, so we currently don't plan to work on this.
> 
> However, given how cross-learning/multivariate forecasting is implemented in Chronos-2, the users can get identical results by simply "melting" the wide DataFrame into a long DataFrame, something like the following
> 
> def multivariate_to_univariate(
>     df: pd.DataFrame,
>     item_id_col: str,
>     time_col: str,
>     target_cols: list[str],
>     target_col: str = "target",
> ) -> pd.DataFrame:
>     out = df.melt(
>         id_vars=[item_id_col, time_col],
>         value_vars=target_cols,
>         var_name="__var",
>         value_name=target_col,
>     )
> 
>     # New item_id will be "{old_item_id}_{target_column}"
>     out[item_id_col] = out[item_id_col].astype(str) + "_" + out["__var"]
> 
>     return (
>         out
>         .drop(columns="__var")
>         .sort_values([item_id_col, time_col])
>         .reset_index(drop=True)
>     )
> 
> This works because with `cross_learning=True`, Chronos-2 does not differentiate between individual univariate items, or multiple multivariate dimensions of a single item. In other words, the following two inputs will produce identical results
> item_id 	timestamp 	X 	Y
> A 	1 	10 	20
> A 	2 	11 	21
> A 	3 	12 	22
> B 	1 	30 	40
> B 	2 	31 	41
> B 	3 	32 	42
> item_id 	timestamp 	target
> A_X 	1 	10
> A_X 	2 	11
> A_X 	3 	12
> A_Y 	1 	20
> A_Y 	2 	21
> A_Y 	3 	22
> B_X 	1 	30
> B_X 	2 	31
> B_X 	3 	32
> B_Y 	1 	40
> B_Y 	2 	41
> B_Y 	3 	42
> 
> In our internal benchmarks we found `batch_size=256` to not result in any noticeable accuracy decrease compared to `batch_size=100`, while resulting in faster inference speed. That's why we decided to keep this value as the default

Thank you for your reply!
